Improve navigation add Quick Start intro (#287)
SpineEventEngine/SpineEventEngine.github.io: docs/guides/validation-user-guide.md

Similarly to `validate()` method in Java, the validation function does not throw exceptions. A list
of `ConstraintViolation`s can be obtained from the `ValidationError`.

## Validation options overview
<a name="validation-options"></a>

In most cases validation constraints are defined for Protobuf message fields such as if a field

> must be populated or its must be withing a range, or have match a regular expression. Not so often

Reviewer: ```suggestion
must be populated or it must be withing a range, or match a regular expression. Not so often
```
